src/sage/crypto/key_exchange/pke.py

```diff
@@ -0,0 +1,41 @@
+r"""
+Public key exchange schemes.
+
+This module contains base classes for public key exchange schemes. The classes defined in
+this module should not be called directly. It is the responsibility of child classes to
+implement specific cryptosystems.
+
+
+AUTHORS:
+
+- Vincent Macri (2024-07-15): initial version
+"""
+# ****************************************************************************
+#       Copyright (C) 2024 Vincent Macri <vincent.macri@ucalgary.ca>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#                  https://www.gnu.org/licenses/
+# ****************************************************************************
+
+from sage.misc.superseded import experimental
+
+class KeyExchangeScheme:
+
+    @experimental(37305)
```

Comment:
This is marked as experimental because if this gets merged I'd like to add more cryptographic schemes to Sage in the near(ish) future as I have spare time. I don't want to be tied down yet to any particular way of structuring the code.